.pre-commit-config.yaml

```diff
@@ -218,39 +218,34 @@ repos:
     - tests/
     pass_filenames: false
 
-- repo: local
+- repo: https://github.com/PyCQA/pylint.git
+  rev: v2.12.2
   hooks:
   - id: pylint
-    language: system
-    name: PyLint
-    files: \.py$
-    entry: python -m pylint
-    args:
-    - --ignore=tm_tokenize
-    - |-
```

Comment:
The removal of this arg could be submitted as a separate PR too.

Reply:
Doesn't the exclude do the same thing?

Reply:
Now I see explained above :)

Reply:
I am fully aware of the problem with immutable dependencies on pre-commit and lack of support for external dependency files. I knew it for years, bit of  PITA but I kinda get used to it and added a separated list of deps inside pylint and mypy configs to deal with it. It is bit duplicating but I think it worth, in the end project dependencies do not change often and even if they change the linter job will be the first to slap us.

Overall I think that the downsides are considerably smaller than the benefits of running under pre-commit as real hooks, not local ones. So far only two hooks needed extra attention with `additional_dependencies`. 

I only had two cleanup pre-commit cache a couple of times in the last two years as usually you can avoid it by altering the additional_dependencies even with a simple extra version condition to avoid resetting entire cache.

I plan to write a blog post about these challenges and maybe get some feedback from Anthony.

Combining unit-test dependencies with linters in a single `test-requirements.txt` is pretty bad for at least two reasons: it bloats all tox environments by installing more than needed and second, it may prevent you from identifying the correct list of dependencies due to cross-contamination.

Also, if we use local hooks, we cannot use them directly with pre-commit (hook or not) and we cannot run them with pre-commit.ci either. I would try to avoid them when possible.

Reply:
It'd like to have a constraints file for this and pre-commit stands in the way of stability if you use non-local hooks.